### PR TITLE
fix: onboard LLMO/ASO to FACS; bypass not-yet-onboarded products (ACO)

### PR DESCRIPTION
**Pull request by ravverma**
## What

Adds `FACS_ONBOARDED_PRODUCTS: ['LLMO', 'ASO']` to `routeFacsCapabilities` and bumps `@adobe/spacecat-shared-http-utils` to **1.35.2** (which ships the product-onboarding gate — adobe/spacecat-shared#1893). `facsWrapper` uses the list to keep LLMO and ASO FACS-enforced while a **recognized-but-not-onboarded** product like **ACO** (`ACO: {}`) bypasses the FACS layer entirely — ACO routes remain governed by the controller-level ACL. `ACO: {}` is left in place: an empty-but-present entry is what marks ACO a *recognized* product (so a bogus `x-product` still fails closed).

## Why

Fixes the `403 {"message":"x-product header required for FACS-governed routes"}` returned on ACO calls that use a Spacecat-signed JWT, e.g.:

```
GET https://aco.experiencecloud.live/api/v1/organizations/by-ims-org-id/<ims-org>   (x-product: ACO)
```

`GET /organizations/by-ims-org-id/:imsOrgId` is declared under LLMO and ASO but `PRODUCTS_ROUTES.ACO` is empty, so the wrapper treated the call as "governed route called with a non-matching product" and denied it. Legacy IMS tokens bypassed FACS (step-1b) and worked; the JWT path does not, which is why the switch to Spacecat-signed JWTs surfaced this.

## Tests

- `test/routes/facs-capabilities.test.js`: `FACS_ONBOARDED_PRODUCTS` added to the strict top-level-keys assertion, plus a validation block (uppercase & unique codes; every onboarded product exists in `PRODUCTS_ROUTES`).
- Suite green (33 passing).

## Dependency

- [x] adobe/spacecat-shared#1893 merged & released as `@adobe/spacecat-shared-http-utils@1.35.2`.
- [x] `@adobe/spacecat-shared-http-utils` bumped `1.35.0` → `1.35.2` in this PR.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Change Management

```yaml
cm-assessment: v1
changeType: standard
impact: unnoticeable
risk: minor
changeApprovedBy: ["Dominique Jäggi"]
rationale: "Auto-added baseline (no PR assessment present at CI time); refine via the cmr skill if this change is higher-impact."
```

### CODE REVIEW

File: test/routes/facs-capabilities.test.js [hunk 1]
```diff
@@ -107,6 +108,30 @@ describe('routeFacsCapabilities', () => {
...
+    });
+
+    it('every onboarded product has a PRODUCTS_ROUTES entry', () => {
+      // A product cannot be enforced by facsWrapper without a route map, and the
+      // wrapper only bypasses *recognized* products, so the two lists must agree.
+      const productKeys = Object.keys(routeFacsCapabilities.PRODUCTS_ROUTES);
+      const orphans = routeFacsCapabilities.FACS_ONBOARDED_PRODUCTS
+        .filter((product) => !productKeys.includes(product));
+      expect(orphans, `onboarded products missing from PRODUCTS_ROUTES: ${orphans.join(', ')}`)
+        .to.deep.equal([]);
```
Comment (MysticatBot) [inline test/routes/facs-capabilities.test.js:131]: issue (blocking): The test validates the forward direction (every onboarded product has a `PRODUCTS_ROUTES` entry) but not the reverse: that a product with a **non-empty** `PRODUCTS_ROUTES` sub-map is listed in `FACS_ONBOARDED_PRODUCTS`.

Today this is harmless because `ACO` has `{}`. But the moment someone adds routes to ACO and forgets to add it to the onboarded list, FACS enforcement is silently bypassed for those routes - the more dangerous failure mode (routes look protected, wrapper never enforces).

Suggested fix - add a third `it()` in this describe block:
```js
it('products with non-empty route maps are onboarded', () => {
  const onboarded = routeFacsCapabilities.FACS_ONBOARDED_PRODUCTS;
  const populated = Object.entries(routeFacsCapabilities.PRODUCTS_ROUTES)
    .filter(([, routes]) => Object.keys(routes).length > 0)
    .map(([product]) => product);
  const unenforced = populated.filter((p) => !onboarded.includes(p));
  expect(unenforced, `products with routes but not onboarded: ${unenforced.join(', ')}`)
    .to.deep.equal([]);
});
```

This closes the gap where a future contributor adds routes without flipping the onboarded flag.

**Review by MysticatBot — CHANGES_REQUESTED** [review]
Hey @ravverma,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Request changes** - one test coverage gap in the authorization enforcement safety net.
**Complexity: HIGH** - medium diff touching API surface + dependencies risk flags.
_Changes: Adds a FACS product-onboarding allowlist and bumps the shared HTTP lib to consume it, fixing 403s on ACO calls using Spacecat-signed JWTs (4 files)._
Note: CI checks are currently pending.

## Must fix before merge
1. [Important] Missing inverse test guard - products with non-empty route maps that are absent from `FACS_ONBOARDED_PRODUCTS` silently bypass FACS enforcement - `test/routes/facs-capabilities.test.js` (details inline)

<details>
<summary>Non-blocking (1): minor issues and suggestions</summary>

- nit: 14-line comment block above `FACS_ONBOARDED_PRODUCTS` is verbose for a config constant; the module-level docstring already covers fail-close semantics - 2-3 lines pointing to the wrapper and stating "add here when routes go live" would suffice - `src/routes/facs-capabilities.js:246`
</details>

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 10m 46s | <b>Cost:</b> $9.44 | <b>Commit:</b> <code>2b224b1815fa78b86f2fe1c04ac9771a59274972</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Review by MysticatBot — APPROVED** [review]
Hey @ravverma,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Approve** - prior findings addressed, no new issues found.
**Complexity: HIGH** - medium diff; API surface + dependencies risk flags.
_Changes: Adds a FACS product-onboarding allowlist and bumps the shared HTTP lib to consume it, fixing 403s on ACO calls using Spacecat-signed JWTs (4 files)._

<details>
<summary>Non-blocking (0): no issues</summary>

No findings.
</details>

## Previously flagged, now resolved
- Inverse test guard added - products with non-empty route maps absent from FACS_ONBOARDED_PRODUCTS now fail the build
- Comment block trimmed from 14 lines to 5, pointing at facsWrapper docs and the test file as enforcement

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 1m 19s | <b>Cost:</b> $6.22 | <b>Commit:</b> <code>788c983acdc5bbeed42c6c41e96253af7c509af4</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>
